src/ui/public/toggle_button/toggle_button.js

```diff
@@ -0,0 +1,21 @@
+import { uiModules } from 'ui/modules';
+import template from './toggle_button.html';
+
+const app = uiModules.get('kibana');
+
+app.directive('toggleButton', function () {
+  return {
+    restrict: 'E',
+    replace: true,
+    template: template,
+    scope: {
+      text: '@',
+      isCollapsed: '=',
+      onClick: '='
```

Comment:
Still more partial on `onToggle`, but as we have no real rules around this stuff, and nobody else seems to care, I'm not going to block the PR on it. This is an improvement.